lib/model.js

```diff
@@ -4665,6 +4664,10 @@ function populate(model, docs, options, callback) {
       populateOptions.strictPopulate = model.base.options.strictPopulate;
     }
   }
+  const originalLocalField = model.schema.virtuals[populateOptions.path].options.localField;
```

Comment:
test failing: `cannot read "options" of undefined`, add a check that that field exists before trying to access it